kube-runtime/src/watcher.rs

```diff
@@ -420,7 +420,8 @@ where
             match api.list(&lp).await {
                 Ok(list) => {
                     objects.extend(list.items);
-                    if let Some(continue_token) = list.metadata.continue_ {
+                    let continue_token = list.metadata.continue_.unwrap_or_default();
+                    if !continue_token.is_empty() {
```

Comment:
I'd just filter the option:

```suggestion
                    if let Some(continue_token) = list.metadata.continue_.filter(|s| !s.is_empty())
```

Reply:
nice, less alloc and less code :+1: